### PR TITLE
Autolight away mode

### LINKED ISSUE #43: D3_Refactoring TimerTask
Refactoring the timerTask in the Dashboard controller
- help tracking the time to set the light on /off during away mode
- update house layout 
- support both timer or count down timer

### PR DESCRIPTION
**Pull request by kimchhengheng**
refactor timerTask inner class method

### CODE REVIEW

**Review by foreversponge — APPROVED** [review]

**Review by BadrSaidi — APPROVED** [review]

File: logs.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Nov 26, 2020 9:31:29 PM models.LogToFileModel log
```
Comment (d-rossi) [inline logs.txt:1]: Do not commit the logs.txt file (this file should remain blank or else logs will already be present in the file)

File: src/main/java/controllers/dashBoardController.java [hunk 2]
```diff
@@ -58,6 +58,16 @@
 	private String selectItem;
 	private List<String> selectLocation;
 
+
+	public Map<String, ArrayList<LocalTime>> getAwayModeLight() {
```
Comment (d-rossi) [inline src/main/java/controllers/dashBoardController.java:62]: Add javadoc for this method

File: src/main/java/controllers/dashBoardController.java [hunk 3]
```diff
@@ -58,6 +58,16 @@
...
+		return awayModeLight;
+	}
+
+	public void setAwayModeLight(Map<String, ArrayList<LocalTime>> awayModeLight) {
```
Comment (d-rossi) [inline src/main/java/controllers/dashBoardController.java:66]: Add javadoc for this method

File: src/main/java/controllers/dashBoardController.java [hunk 4]
```diff
@@ -68,19 +78,68 @@
...
+			this.mode = mode;
+		}
+
 		public void setTimeInc(int timeInc) {
 			this.timeInc = timeInc;
 		}
 		private void setTime(LocalTime ctime) {
 			this.localTime = ctime;
 		}
+		private void awayModeLight(){
```
Comment (d-rossi) [inline src/main/java/controllers/dashBoardController.java:96]: Add javadoc for this method

File: src/main/java/controllers/dashBoardController.java [hunk 5]
```diff
@@ -68,19 +78,68 @@
...
+							}
+						}
+					}
+				}
+			}
+			if(update){
+				displayLayout();
+			}
+		}
+		private void increment(){
```
Comment (d-rossi) [inline src/main/java/controllers/dashBoardController.java:123]: Add javadoc for this method

File: src/main/java/controllers/dashBoardController.java [hunk 6]
```diff
@@ -68,19 +78,68 @@
...
+			localTime = localTime.plusSeconds(timeInc);
+			time.setText(localTime.format(DateTimeFormatter.ofPattern("HH:mm:ss")));
+			awayModeLight();
+		}
+		private void decrement(){
```
Comment (d-rossi) [inline src/main/java/controllers/dashBoardController.java:128]: Add javadoc for this method

File: logs.txt (hunk 1 above)
Comment (kimchhengheng) [reply logs.txt:1]: i do not change this part, it was from the previous delivery

**Pull request closed without merging**